src/dbnode/persist/fs/retriever.go

```diff
@@ -607,6 +721,23 @@ type retrieveRequest struct {
 	notFound bool
 }
 
+func (req *retrieveRequest) onIndexChecksumCompleted(
+	indexChecksum ident.IndexChecksum) {
+	if req.err == nil {
+		req.indexChecksum = indexChecksum
+		// If there was an error, we've already called done.
```

Comment:
Maybe I missed it but where does this happen?

Reply:
This happens when `req.onError(err)` is called within `fetchBatch` during processing this, which will set the err to nil and release the wg